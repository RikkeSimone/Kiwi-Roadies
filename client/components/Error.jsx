import React from 'react'
import { connect } from 'react-redux'

export const Error = (props) => {
  return (
    <>
      { props.errorMessage &&
      <div className='error'>
        <p>error message placeholder</p>
        {props.errorMessage}
      </div>
      }
    </>
  )
}

function mapStateToProps (state) {
  return {
    errorMessage: state.error
  }
}

export default connect(mapStateToProps)(Error)
