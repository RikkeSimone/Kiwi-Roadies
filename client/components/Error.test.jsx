import React from 'react'
import { renderWithRedux } from '../testing/utils'
import '@testing-library/jest-dom'
// To resolve the ReferenceError: regeneratorRuntime is not defined error, we need to import this runtime module
/* eslint-disable no-unused-vars */
import regeneratorRuntime from 'regenerator-runtime'
/* eslint-enable no-unused-vars */
import Error from './Error'
// const initialState = {
//   error: null
// }

const newState = {
  error: 'error'
}

// test('Error component for falsey for initial store value and when there is no error', async () => {
//   const { asFragment } = renderWithRedux(<Error />, { initialState })
//   expect(asFragment()).toMatchSnapshot()
// })

test('Error component for truthy when store changes', async () => {
  const { asFragment } = renderWithRedux(<Error />, { newState })
  expect(asFragment()).toMatchSnapshot()
})
