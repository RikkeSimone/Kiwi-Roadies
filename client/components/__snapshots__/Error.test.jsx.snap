// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component for falsey for initial store value and when there is no error 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <p>
      error message placeholder
    </p>
  </div>
</DocumentFragment>
`;

exports[`Error component for truthy when store changes 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <p>
      error message placeholder
    </p>
  </div>
</DocumentFragment>
`;
